### ISSUE TITLE
(4.0.2) caused by: Cannot read property 'run' of undefined ; Zone: angular ; Task: Promise.then

### ISSUE THREAD

**Issue by alanpurple** [swimlane/ngx-charts#200]
caused by: Cannot read property 'run' of undefined ; Zone: angular ; Task: Promise.then

check your demo

occur with both angular 2.4.5 and 4.0.0-beta.5

**Comment by alanpurple** [comment]
this doesn't happen with 4.0.1

**Comment by mariuspruski** [comment]
I think my issue is similar:
`EXCEPTION: Error in ./NgXBarChartComponent class NgXBarChartComponent - inline template:11:2 caused by: this.zone is undefined`

where NgXBarChartComponent is a simple wrapper:
```
<ngx-charts-bar-horizontal
  [results]="data"
  [view]="size"
  [scheme]="scheme"
  [schemeType]="schemeType"
  [xAxis]="showXAxis"
  [yAxis]="showYAxis"
  [xAxisLabel]="xAxisLabel"
  [showXAxisLabel]="showXAxisLabel"
  [showGridLines]="showGridLines"
  [gradient]="gradient"
  [activeEntries]="activeEntries"
  [xAxisTickFormatting]="xAxisTickFormatting"
  [roundDomains]="roundDomains"
  [barPadding]="barPadding"
>
</ngx-charts-bar-horizontal>
```

```
import {Component, OnInit, Input} from '@angular/core';

@Component({
  selector: 'ngx-bar-chart',
  templateUrl: './ngx-bar-chart.component.html',
  styleUrls: ['./ngx-bar-chart.component.css']
})
export class NgXBarChartComponent implements OnInit {
  @Input() data;
  roundDomains = true;
  showGridLines = true;
  gradient = false;
  barPadding = 8;
  activeEntries = [];
  size = [300, 200];
  showXAxis = true;
  showYAxis = true;
  showXAxisLabel = true;
  xAxisLabel = "Number of documents";
  xAxisTickFormatting = function(d) {
    if (Number.isInteger(d)) return String(d);
    return "";
  };
  scheme = {
    name: 'vivid',
    selectable: true,
    group: 'Ordinal',
    domain: [
      '#647c8a', '#3f51b5', '#2196f3', '#00b862', '#afdf0a', '#a7b61a', '#f3e562', '#ff9800', '#ff5722', '#ff4514'
    ]
  };
  schemeType='ordinal';
  ngOnInit(): void {}
  constructor() { }
}
```

This code ran fine in 4.0.1. Since I'm not using `this.zone`, I'm assuming something broke there?

**Comment by marjan-georgiev** [comment]
Released a fix in 4.0.3. Please let me know if it works. We updated angular to 2.4.5 in 4.0.2 without updating the zone.js version.

**Comment by alanpurple** [comment]
@marjan-georgiev  works well
